test-cloud/test-cloud-performance/src/test/java/org/kie/cloud/integrationtests/s2i/JbpmEJBInterleavedTimersPerfIntegrationTest.java

```diff
@@ -0,0 +1,111 @@
+    
+    private static final int BATCH_COUNT = Integer.parseInt(System.getProperty("batchCount", "5"));
+    private static final int BATCH_SIZE = PROCESSES_COUNT / BATCH_COUNT;
+    private static final int BATCH_DELAY = Integer.parseInt(System.getProperty("batchDelay", "10"));
+    private static final int TIMER_DELAY = Integer.parseInt(System.getProperty("timerDelay", "1"));
+    private static final int BATCH_MAX_TIME = Integer.parseInt(System.getProperty("batchMaxTime", "20"));
+    
+    protected static final int STARTING_THREADS_COUNT = 20;
+
+    protected static final int PROCESSES_PER_THREAD = BATCH_SIZE / STARTING_THREADS_COUNT;
+    
+    protected String[] startingTime = new String[BATCH_COUNT];
+    protected String[] processTime = new String[BATCH_COUNT];
+    
+    @Override
+    protected void writeCSV() throws IOException {
+        ArrayList<Object> record = new ArrayList<Object>();
+        try (
+               CSVPrinter csvPrinter = new CSVPrinter(new FileWriter(Paths.get(CSV_FILE).toAbsolutePath().toFile(), true), 
+                        CSVFormat.DEFAULT);
+            ) {
+             record.add(Instant.now());
+             record.add(PROCESSES_COUNT); 
+             record.add(STARTING_THREADS_COUNT);
+             for (int i = 0; i < BATCH_COUNT; i++) {
+                 logger.info("Recording batch no. {}:  {} - {}", i, startingTime[i], processTime[i]);
+                 record.add(startingTime[i].substring(2));
+                 record.add(processTime[i].substring(2));
+             }
+             record.add(HEAP);
+             record.addAll(completedHostNameDistribution.values());
+             csvPrinter.printRecord(record);
+             csvPrinter.flush();            
+            }
+        
+    }
+
+    @Override
+    protected void runSingleScenario() {
+        logger.info("Starting {} batches", BATCH_COUNT);
+        for (int i = 0; i < BATCH_COUNT;) {
+            logger.info("Starting batch no. {}", i);
+
+            logger.info("Starting {} processes", BATCH_SIZE);
+
+            Instant startTime = Instant.now();
+            // This is just to provide virtually "infinite" time to finish all iterations, i.e. maximum possible duration minus 1 hour,
+            // so we can be sure there won't be overflow
+            Duration maxDuration = Duration.between(startTime.plus(1, ChronoUnit.HOURS), Instant.MAX);
+            Map<String, Object> params = Collections.singletonMap("timerDelay", TIMER_DELAY);
+
+            startAndWaitForStartingThreads(STARTING_THREADS_COUNT, maxDuration, PROCESSES_PER_THREAD, getStartingRunnable(CONTAINER_ID, ONE_TIMER_DURATION_PROCESS_ID, params));
+            
+            startingTime[i] = Duration.between(startTime, Instant.now()).toString();
+            logger.info("Starting processes took: {}", startingTime[i]);
+
+            Duration waitForCompletionDuration = Duration.of(BATCH_MAX_TIME, ChronoUnit.MINUTES);
```

Comment:
So here we basically wait until all timers are fired and all processes from that batch finished, don't we?

Reply:
yes, in the _waitForAllProcessesToComplete_ method, 2 lines below.

Reply:
So I assume that 20 minutes is enough then for all tested sample sizes.